diff --git a/numpy/f2py/setup.py b/numpy/f2py/setup.py
index 3204129..0d47580 100644
--- a/numpy/f2py/setup.py
+++ b/numpy/f2py/setup.py
@@ -55,9 +55,9 @@ def configuration(parent_package='', top_path=None):
     config.make_svn_version_py()
 
     def generate_f2py_py(build_dir):
-        f2py_exe = 'f2py' + os.path.basename(sys.executable)[6:]
-        if f2py_exe[-4:] == '.exe':
-            f2py_exe = f2py_exe[:-4] + '.py'
+        f2py_exe = 'f2py' + str(sys.version_info.major)
+        if sys.executable.endswith('.exe'):
+            f2py_exe = f2py_exe + '.py'
         if 'bdist_wininst' in sys.argv and f2py_exe[-3:] != '.py':
             f2py_exe = f2py_exe + '.py'
         target = os.path.join(build_dir, f2py_exe)
